sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -181,6 +217,142 @@ class GpuWindowExecMeta(windowExec: WindowExec,
 }
 
 object GpuWindowExec extends Arm {
+  /**
+   * As a part of `splitAndDedup` the dedup part adds a layer of indirection. This attempts to
+   * remove that layer of indirection.
+   * @param windowOps the windowOps output of splitAndDedup
+   * @param post the post output of splitAndDedup
+   * @return a version of windowOps that has removed as many un-needed temp aliases as possible.
+   */
+  def remapAttributes(windowOps: Seq[NamedExpression],
+      post: Seq[NamedExpression]): Seq[NamedExpression] = {
+    val postRemapping = post.flatMap {
+      case a @ GpuAlias(attr: AttributeReference, _) => Some((attr.exprId, a))
+      case _ => None
+    }.groupBy(_._1)
+    windowOps.map {
+      case a @ GpuAlias(child, _)
+        // We can only replace the mapping if there is one thing to map it to.
+        if postRemapping.get(a.exprId).exists(_.length == 1) =>
+        val attr = postRemapping(a.exprId).head._2
+        GpuAlias(child, attr.name)(attr.exprId, attr.qualifier)
+      case other => other
+    }
+  }
+
+  private def hasGpuWindowFunction(expr: Expression): Boolean = {
+    expr.find {
+      case _: GpuWindowExpression => true
+      case _ => false
+    }.isDefined
+  }
+
+  private def extractAndSave(expr: Expression,
+      saved: ArrayBuffer[NamedExpression],
+      deduped: mutable.HashMap[Expression, Attribute]): Expression =
+    expr match {
+      // Don't rename an already named expression
+      case ne: NamedExpression =>
+        val missingExprId = Set(ne.exprId) -- saved.map(_.exprId)
+        if (missingExprId.nonEmpty) {
+          saved += ne
+        }
```

Comment:
Nit: I found the set operations on set-of-1 to be a bit odd.  The following has the benefit of early out:
```suggestion
        if (!saved.exists(_.exprId == ne.exprId)) {
          saved += ne
        }
```